test/cypress/integration/worldpay-3ds-flex-ddc/payment-with-session-id.spec.js

```diff
@@ -0,0 +1,138 @@
+    { name: 'connectorValidPatchConfirmedChargeDetails', opts: { chargeId } }
+  ]
+
+  const worldpay3dsFlexDdcStub = {
+    name: 'worldpay3dsflexddcIframePost', opts: { sessionId }
+  }
+
+  beforeEach(() => {
+    // this test is for the full process, the session should be maintained
+    // as it would for an actual payment flow
+    Cypress.Cookies.preserveOnce('frontend_state')
+  })
+
+  const setUpAndCheckCardPaymentPage = () => {
+    describe('Secure card payment page', () => {
+      it('Should setup the payment and load the page', () => {
+        cy.task('setupStubs', createPaymentChargeStubsEnglish)
+        cy.visit(`/secure/${tokenId}`)
+
+        // 1. Charge will be created using this id as a token (GET)
+        // 2. Token will be deleted (DELETE)
+        // 3. Charge will be fetched (GET)
+        // 4. Service related to charge will be fetched (GET)
+        // 5. Charge status will be updated (PUT)
+        // 6. Client will be redirected to /card_details/:chargeId (304)
+        cy.location('pathname').should('eq', `/card_details/${chargeId}`)
+      })
+
+      it('Should enter and validate a correct card', () => {
+        cy.task('setupStubs', checkCardDetailsStubs)
+
+        cy.server()
+        cy.route('POST', `/check_card/${chargeId}`).as('checkCard')
+
+        cy.get('#card-no').type(validPayment.cardNumber)
+        cy.get('#card-no').blur()
+
+        // 7. Charge status will be fetched (GET)
+        // 8. CardID POST to verify that entered card is correct - this is configured to return valid
+        cy.wait('@checkCard')
+        cy.get('#card-no').should('not.have.class', 'govuk-input--error')
+      })
+
+      it('Should enter payment details', () => {
+        cy.get('#expiry-month').type(validPayment.expiryMonth)
+        cy.get('#expiry-year').type(validPayment.expiryYear)
+        cy.get('#cardholder-name').type(validPayment.name)
+        cy.get('#cvc').type(validPayment.securityCode)
+        cy.get('#address-line-1').type(validPayment.addressLine1)
+        cy.get('#address-city').type(validPayment.city)
+        cy.get('#address-postcode').type(validPayment.postcode)
+        cy.get('#email').type(validPayment.email)
+      })
+    })
+  }
+
+  setUpAndCheckCardPaymentPage()
+  describe('Secure confirmation page', () => {
+    it('Submitting confirmation with valid details and a JWT present should cause worldpay iFrame to pass extra input when submitting with the session ID', () => {
+      cy.task('setupStubs', [...confirmPaymentDetailsStubs, worldpay3dsFlexDdcStub])
+
+      cy.get('#card-details').submit().should($form => {
+        const formVal = $form.first()[0].elements.worldpay3dsFlexDdcResult.value
+        expect(formVal).to.eq(sessionId)
+      })
+    })
+  })
+
+  setUpAndCheckCardPaymentPage()
+  describe('Secure confirmation page', () => {
+    it('Submitting confirmation when Worldpay does not respond to iframe post should submit the form without the sessionId', () => {
```

Comment:
I think having “times out” (or similar) in here would be make it more descriptive

Reply:
yep 👍